src/robot_interface/models/mission/step.py

```diff
@@ -15,7 +15,7 @@ class Step:
     Base class for all steps in a mission.
     """
 
-    id: UUID = field(default_factory=uuid4, init=False)
+    id: Union[UUID, str] = field(default_factory=uuid4, init=False)
```

Comment:
Should we change to only use str? I think Union makes for confusion and I dont see the point. Imo we just change the UUID to a str as soon as it is created, then keep it str